code/hud/hudlock.cpp

```diff
@@ -835,7 +835,10 @@ void hud_lock_acquire_uncaged_target(lock_info *current_lock, weapon_info *wip)
 	int least_num_locks = INT_MAX;
 	bool actively_locking = false;
 
-	for ( A = GET_FIRST(&obj_used_list); A !=END_OF_LIST(&obj_used_list); A = GET_NEXT(A) ) {
+	for ( auto so = GET_FIRST(&Ship_obj_list); so != END_OF_LIST(&Ship_obj_list); so = GET_NEXT(so) ) {
```

Comment:
Good fix.  Can we use either so or sp, since we don't need both?

Reply:
We can't, unfortunately.  The `so` variable is of the `ship_object` type, which is distinct from both the `object` and `ship` types.

What I could do is give it a `ship_obj` name to make that a little more clear.

Reply:
Cancel that... `ship_obj` is the type itself, not `ship_object`.  And `so` is the iterator variable used pretty much everywhere.

Reply:
Actually, there's a better fix here.